File: packages/core/tests/fixtures/javascript/code/modules/uses_user.js
/**
 * Cross-module class usage - uses_user.js
 * Tests: ES6 class imports, constructor calls, method calls across modules
 */

import { User } from './user_class.js';

// Constructor call and type binding
const user = new User("Alice Smith", "alice@example.com");

// Method calls on imported class instance
const userName = user.getName();
const userEmail = user.getEmail();

// Method chaining
const updatedUser = user.updateProfile("Alice Johnson", "alice.johnson@example.com");

// Additional method call after update
const finalName = updatedUser.getName();
const userInfo = user.getInfo();

// Export results
export {
  user,
  userName,
  userEmail,
  finalName,
  userInfo,
};

File: packages/core/tests/fixtures/python/code/modules/uses_user.py
"""
Cross-module class usage - uses_user.py
Tests: Python class imports, constructor calls, method calls across modules
"""

from .user_class import User

# Constructor call and type binding
user = User("Alice Smith", "alice@example.com")

# Method calls on imported class instance
user_name = user.get_name()
user_email = user.get_email()

# Method chaining
updated_user = user.update_profile(name="Alice Johnson", email="alice.johnson@example.com")

# Additional method calls after update
final_name = updated_user.get_name()
user_info = user.get_info()

# Method call with side effects
user.deactivate()
final_info = user.get_info()

# Store results
results = {
    "user": user,
    "user_name": user_name,
    "user_email": user_email,
    "final_name": final_name,
    "user_info": user_info,
    "final_info": final_info,
}